SystemServices/SystemServices.cpp

```diff
@@ -2110,82 +2112,53 @@ namespace WPEFramework {
             sendNotify(EVT_ONTEMPERATURETHRESHOLDCHANGED, params);
         }
 
-        /***
-         * @brief : To set the Time to TZ_FILE.
-         * @param1[in]	: {"params":{"timeZone":"<string>"}}
-         * @param2[out]	: {"jsonrpc":"2.0","id":3,"result":{"success":<bool>}}
-         * @return		: Core::<StatusCode>
-         */
         uint32_t SystemServices::setTimeZoneDST(const JsonObject& parameters,
-                JsonObject& response)
-	{
-		bool resp = false;
-		if (parameters.HasLabel("timeZone")) {
-			std::string dir = dirnameOf(TZ_FILE);
-			std::string timeZone = "";
-			try {
-				timeZone = parameters["timeZone"].String();
-				if (timeZone.empty() || (timeZone == "null")) {
-					LOGERR("Empty timeZone received.");
-				} else {
-					if (!dirExists(dir)) {
-						std::string command = "mkdir -p " + dir + " \0";
-						Utils::cRunScript(command.c_str());
-					} else {
-						//Do nothing//
-					}
-
-					FILE *f = fopen(TZ_FILE, "w");
-					if (f) {
-						if (timeZone.size() != fwrite(timeZone.c_str(), 1, timeZone.size(), f))
-							LOGERR("Failed to write %s", TZ_FILE);
-
-						fflush(f);
-						fsync(fileno(f));
-						fclose(f);
-						resp = true;
-					} else {
-						LOGERR("Unable to open %s file.\n", TZ_FILE);
-						populateResponseWithError(SysSrv_FileAccessFailed, response);
-						resp = false;
-					}
-				}
-			} catch (...) {
-				LOGERR("catch block : parameters[\"timeZone\"]...");
-			}
-		} else {
-			populateResponseWithError(SysSrv_MissingKeyValues, response);
-		}
-		returnResponse(resp);
-	}
+            JsonObject& response)
+        {
+            bool resp = false;
+
+            if (parameters.HasLabel("timeZone")) {
+                string timeZone = parameters["timeZone"].String();
+
+                if (timeZone.empty() || (timeZone == "null")) {
+                    LOGERR("Empty timeZone received.");
+                } else {
+                    auto service = m_shellService->
+                        QueryInterfaceByCallsign<Exchange::ITimeZone>("LocationSync");
+
+                    if ((service != nullptr) &&
+                        (service->SetTimeZone(timeZone) == Core::ERROR_NONE)) {
```

Comment:
Pls add support to transfer any saved timezones from the legacy location (/opt/persistent/timeZoneDST) to avoid regression while updating builds.